### PR TITLE
Add gregor tlf notifs desktop 1598

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Sets us up for to receive gregor messages and does it through sagas

### CODE REVIEW

File: shared/actions/gregor.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    const {payload: {state}} = pushAction
+    // Check if any are a tlf items
+    const tlfUpdates = toNonNullGregorItems(state).filter(isTlfItem)
+    const newTlfUpdates = tlfUpdates.filter(gItem => !seenMsgs[gItem.md.msgID.toString('base64')])
+    if (newTlfUpdates.length) {
+      yield put(updateSeenMsgs(newTlfUpdates))
+      yield put(favoriteList())
+    }
+  } else {
+    console.log('Error in gregor pushState', pushAction.payload)
```
Comment (chrisnojima) [inline shared/actions/gregor.js]: shouldn't this error handling happen outside of this? for example if we listen to another type of gregor notification (coming soon) then this pushAction will go to some other handler also, lets say handleChatUpdate, then if there's an error it'll be logged twice?

File: shared/actions/gregor.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+function * handleTLFUpdate (pushAction: PushState): SagaGenerator<any, any> {
+  // $ForceType
+  let seenMsgs: MsgMap = yield select((state: TypedState) => state.gregor.seenMsgs)
+
+  if (!pushAction.error) {
+    const {payload: {state}} = pushAction
+    // Check if any are a tlf items
+    const tlfUpdates = toNonNullGregorItems(state).filter(isTlfItem)
```
Comment (chrisnojima) [inline shared/actions/gregor.js]: shouldn't this also happen at a higher level. why don't we just filter this out when we get the messages in the first place and then we don't even need this type (they'll always be non-null when they're past the  incomingRpc

File: shared/reducers/gregor.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        const newMsgs: MsgMap = keyBy(action.payload.seenMsgs, m => m.md.msgID.toString('base64'))
+        return {
+          ...state,
+          seenMsgs: {
+            ...state.seenMsgs,
+            ...newMsgs,
+          },
+        }
+      }
+      return state
```
Comment (chrisnojima) [inline shared/reducers/gregor.js]: i know we do this elsewhere but maybe the pattern should always have a return state at the bottom and we just break from the switch statement

File: shared/actions/gregor.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/actions/gregor.js]: yep, I was just avoiding making a parent function to accept the action until we need it. But I can change

**Comment by MarcoPolo** [comment]
@chrisnojima bump

**Comment by chrisnojima** [comment]
👍

**Pull request merged**